apm-lambda-extension/extension/logger_test.go

```diff
@@ -0,0 +1,90 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package extension
+
+import (
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap/zapcore"
+	"io/ioutil"
+	"os"
+	"testing"
+)
+
+func init() {
+	os.Unsetenv("ELASTIC_APM_LOG_LEVEL")
+}
+
+func TestInitLogger(t *testing.T) {
+	assert.NotNil(t, Log)
+}
+
+func TestDefaultLogger(t *testing.T) {
+	tempFile, err := ioutil.TempFile(os.TempDir(), "tempFileLoggerTest-")
+	require.NoError(t, err)
+	defer tempFile.Close()
+
+	SetLogOutputPaths([]string{tempFile.Name()})
+	defer SetLogOutputPaths([]string{"stderr"})
+
+	Log.Infof("%s", "logger-test-info")
+	Log.Debugf("%s", "logger-test-debug")
+	tempFileContents, err := ioutil.ReadFile(tempFile.Name())
+	require.NoError(t, err)
+	assert.Regexp(t, `{"log.level":"info","@timestamp":".*","log.origin":{"file.name":"extension/logger_test.go","file.line":.*},"message":"logger-test-info","ecs.version":"1.6.0"}`, string(tempFileContents))
+}
+
+func TestLoggerParseLogLevel(t *testing.T) {
+	trace, _ := ParseLogLevel("TRacE")
+	debug, _ := ParseLogLevel("dEbuG")
+	info, _ := ParseLogLevel("InFo")
+	warn, _ := ParseLogLevel("WaRning")
+	errorL, _ := ParseLogLevel("eRRor")
+	critical, _ := ParseLogLevel("CriTicaL")
+	off, _ := ParseLogLevel("OFF")
+	invalid, err := ParseLogLevel("Inva@Lid3")
+	assert.Equal(t, zapcore.DebugLevel, trace)
+	assert.Equal(t, zapcore.DebugLevel, debug)
+	assert.Equal(t, zapcore.InfoLevel, info)
+	assert.Equal(t, zapcore.WarnLevel, warn)
+	assert.Equal(t, zapcore.ErrorLevel, errorL)
+	assert.Equal(t, zapcore.PanicLevel, critical)
+	assert.Equal(t, zapcore.FatalLevel, off)
+	assert.Equal(t, zapcore.InfoLevel, invalid)
+	assert.Error(t, err, "invalid log level string Inva@Lid3")
+}
+
+func TestLoggerSetLogLevel(t *testing.T) {
+	tempFile, err := ioutil.TempFile(os.TempDir(), "tempFileLoggerTest-")
+	require.NoError(t, err)
+	defer tempFile.Close()
```

Comment:
```suggestion
	defer os.RemoveAll(tempFile.Name())
	defer tempFile.Close()
```

From the docs:

> It is the caller's responsibility to remove the file when no longer needed.

Alternatively you could use [`testing.T.TempDir`](https://pkg.go.dev/testing#T.TempDir), which is automatically removed at the end of the test.

Reply:
I ended up using `testing.T.TempDir` in 508175e